SDWebImage/SDAnimatedImage.m

```diff
@@ -0,0 +1,158 @@
+/*
+ * This file is part of the SDWebImage package.
+ * (c) Olivier Poitrey <rs@dailymotion.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+#import "SDAnimatedImage.h"
+#import "UIImage+WebCache.h"
+#import "SDWebImageCoder.h"
+#import "SDWebImageCodersManager.h"
+
+static CGFloat SDImageScaleFromPath(NSString *string) {
+    if (string.length == 0 || [string hasSuffix:@"/"]) return 1;
+    NSString *name = string.stringByDeletingPathExtension;
+    __block CGFloat scale = 1;
+    
+    NSRegularExpression *pattern = [NSRegularExpression regularExpressionWithPattern:@"@[0-9]+\\.?[0-9]*x$" options:NSRegularExpressionAnchorsMatchLines error:nil];
+    [pattern enumerateMatchesInString:name options:kNilOptions range:NSMakeRange(0, name.length) usingBlock:^(NSTextCheckingResult *result, NSMatchingFlags flags, BOOL *stop) {
+        if (result.range.location >= 3) {
+            scale = [string substringWithRange:NSMakeRange(result.range.location + 1, result.range.length - 2)].doubleValue;
+        }
+    }];
+    
+    return scale;
+}
+
+@interface SDAnimatedImage ()
+
+@property (nonatomic, strong) id<SDWebImageAnimatedCoder> coder;
+@property (nonatomic, assign, readwrite) NSUInteger animatedImageLoopCount;
+@property (nonatomic, assign, readwrite) NSUInteger animatedImageFrameCount;
+@property (nonatomic, assign, readwrite) SDImageFormat animatedImageFormat;
+@property (nonatomic, assign) BOOL animatedImageLoopCountCheck;
+@property (nonatomic, assign) BOOL animatedImageFrameCountChecked;
+
+#if SD_MAC
+@property (nonatomic, assign) CGFloat scale;
+#endif
+
+@end
+
+@implementation SDAnimatedImage
+
+#pragma mark - UIImage override method
++ (instancetype)imageWithContentsOfFile:(NSString *)path {
+    return [[self alloc] initWithContentsOfFile:path];
+}
+
++ (instancetype)imageWithData:(NSData *)data {
+    return [[self alloc] initWithData:data];
+}
+
++ (instancetype)imageWithData:(NSData *)data scale:(CGFloat)scale {
+    return [[self alloc] initWithData:data scale:scale];
+}
+
+- (instancetype)initWithContentsOfFile:(NSString *)path {
+    NSData *data = [NSData dataWithContentsOfFile:path];
+    return [self initWithData:data scale:SDImageScaleFromPath(path)];
+}
+
+- (instancetype)initWithData:(NSData *)data {
+    return [self initWithData:data scale:1];
+}
+
+- (instancetype)initWithData:(NSData *)data scale:(CGFloat)scale {
+    @autoreleasepool {
```

Comment:
Wouldn't it be more reasonable to put `@autoreleasepool` inside the loop?

Reply:
Sounds good. I miss this